test_conformance/api/negative_queue.cpp

```diff
@@ -0,0 +1,145 @@
+
+    test_failure_error_ret(
+        test_error, CL_INVALID_QUEUE_PROPERTIES,
+        "clCreateCommandQueue should return CL_INVALID_QUEUE_PROPERTIES if "
+        "values specified in properties are valid but are not supported by "
+        "the "
+        "device.",
+        TEST_FAIL);
+    return TEST_PASS;
+}
+
+int test_negative_create_command_queue_with_properties(cl_device_id deviceID,
+                                                       cl_context context,
+                                                       cl_command_queue queue,
+                                                       int num_elements)
+{
+    cl_command_queue_properties device_props = 0;
+    cl_int error = clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_PROPERTIES,
+                                   sizeof(device_props), &device_props, NULL);
+    test_error(error, "clGetDeviceInfo for CL_DEVICE_QUEUE_PROPERTIES failed");
+
+    cl_command_queue_properties device_on_host_props = 0;
+    error = clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_ON_HOST_PROPERTIES,
+                            sizeof(device_on_host_props), &device_on_host_props,
+                            NULL);
+    test_error(error,
+               "clGetDeviceInfo for CL_DEVICE_QUEUE_ON_HOST_PROPERTIES failed");
+
+    if (device_on_host_props != device_props)
+    {
+        log_error(
+            "ERROR: CL_DEVICE_QUEUE_PROPERTIES and "
+            "CL_DEVICE_QUEUE_ON_HOST_PROPERTIES properties should match\n");
+        return TEST_FAIL;
+    }
+
+    // CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE is the only optional host-queue
+    // property to clCreateCommandQueueWithProperties,
+    // CL_QUEUE_PROFILING_ENABLE is mandatory.
+    const bool out_of_order_device_support =
+        device_props & CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE;
+    if (out_of_order_device_support)
+    {
+        // Early return as we can't check correct error is returned for
+        // unsupported property.
+        return TEST_PASS;
+    }
+
+    // Try create a command queue with out-of-order property and check return
+    // code
+    cl_command_queue_properties queue_prop_def[] = {
+        CL_QUEUE_PROPERTIES, CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE, 0
+    };
+
+    cl_int test_error = CL_SUCCESS;
+    clCommandQueueWrapper test_queue = clCreateCommandQueueWithProperties(
+        context, deviceID, queue_prop_def, &test_error);
+
+    test_failure_error_ret(test_error, CL_INVALID_QUEUE_PROPERTIES,
+                           "clCreateCommandQueueWithProperties should "
+                           "return CL_INVALID_QUEUE_PROPERTIES if "
+                           "values specified in properties are valid but "
+                           "are not supported by the "
+                           "device.",
+                           TEST_FAIL);
+
+    // Verify the same correct error reporting for
+    // clCreateCommandQueueWithPropertiesKHR
+    if (is_extension_available(deviceID, "cl_khr_create_command_queue"))
```

Comment:
This isn't a must-fix, since most implementations have moved to OpenCL 3.0 and hence this won't make much of a difference in practice, but: it might be worth moving this extension part to its own test since it can run with any OpenCL version, whereas the core version requires OpenCL 2.0 or newer.

Reply:
Good shout, I think that makes sense - update to have the KHR in its own test.